s3/src/main/scala/geotrellis/store/s3/util/S3RangeReader.scala

```diff
@@ -36,22 +36,29 @@ import java.net.URI
  */
 class S3RangeReader(
   request: => GetObjectRequest,
-  client: S3Client
+  client: S3Client,
+  readHeader: Boolean = true
```

Comment:
This should default to `false` just like `HttpRangeHeader`. Since this problem was encountered while reading one of the AWS public datasets (which relates to most of the base use cases for GeoTrellis) being able to handle that limited permission on some of those buckets should be considered the default.